packages/headless/src/features/commerce/context/cart/cart-analytics-actions.ts

```diff
@@ -0,0 +1,52 @@
+import {CurrencyCodeISO4217, Ec} from '@coveo/relay-event-types';
+import {CartItem} from '../../../../controllers/commerce/context/cart/headless-cart';
+import {itemSelector} from '../../../../controllers/commerce/context/cart/headless-cart-selectors';
+import {
+  CartAction,
+  makeAnalyticsAction,
+} from '../../../analytics/analytics-utils';
+import {CartItemWithMetadata} from './cart-state';
+
+export const logCartAction = (item: CartItem): CartAction =>
+  makeAnalyticsAction({
+    prefix: 'analytics/cart/cartAction',
+    __legacy__getBuilder: (_client, _state) => null,
+    analyticsType: 'ec.cartAction',
+    analyticsPayloadBuilder: (state): Ec.CartAction => {
+      const prevItem = itemSelector(state.cart, item.productId);
+      const action = isCurrentItemQuantityGreater(item, prevItem)
+        ? 'add'
+        : 'remove';
+      const {quantity, ...product} = item;
+      const currency =
+        state.commerceContext.currency.toUpperCase() as CurrencyCodeISO4217;
```

Comment:
I think it would make sense. The question would be if we should make this change with this branch (I guess it could be a breaking change if we change the type?)

Reply:
A separate PR makes sense. Would it be possible to remove the `.toUpperCase` and cast in this PR though? Or are they needed?

Reply:
They are needed, but I could add a function to check if the currency is `CurrencyCodeISO4217` :eyes: . If the currency is not a `CurrencyCodeISO4217` I think I couldn't send the event

Reply:
Changing the behavior would tie to [this comment](https://github.com/coveo/ui-kit/pull/3630#discussion_r1504472576). If actions normally send the event event if invalid, I think we need to do the same. Since the cast is needed, let's keep it (we can remove the `toUpperCase`), and add a jira to tackle it in a follow-up PR.

Reply:
[Jira](https://coveord.atlassian.net/browse/LENS-1589) to do the switch!